ElectronicObserver/Window/Control/Paging/PagingControlViewModel.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+
+namespace ElectronicObserver.Window.Control.Paging;
+
+public partial class PagingControlViewModel : ObservableObject
+{
+	public int CurrentPage { get; set; } = 0;
+
+	private ObservableCollection<object?> Items { get; set; } = new();
+
+	public List<object?> ItemsPaged { get; set; } = new();
+
+	public int ItemsPerPage { get; set; } = 10;
+
+	public int LastPage => (int)Math.Ceiling(Items.Count / (decimal)ItemsPerPage);
+
+	public PagingControlTranslationViewModel PagingControl { get; set; } = new();
+
+	public PagingControlViewModel()
+	{
+		PropertyChanged += OnPagerUpdate;
+		PropertyChanged += OnPagerUpdate2;
+		UpdateCollection();
+	}
+
+	public PagingControlViewModel(ObservableCollection<object?> items) : this()
+	{
+		Items = items;
+		Items.CollectionChanged += Items_CollectionChanged;
+	}
+
+	private void OnPagerUpdate2(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+	{
+		if (e.PropertyName is nameof(CurrentPage) or nameof(LastPage)) return;
+
+		NextPageCommand.NotifyCanExecuteChanged();
+		PreviousPageCommand.NotifyCanExecuteChanged();
+	}
+
+
+	private void Items_CollectionChanged(object? sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+	{
+		UpdateCollection();
+	}
+
+	private void OnPagerUpdate(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+	{
+		if (e.PropertyName is not nameof(CurrentPage) and not nameof(ItemsPerPage)) return;
+
+		UpdateCollection();
+	}
+
+	private void UpdateCollection()
+	{
+		// dont go out of bound
+		CurrentPage = Math.Max(CurrentPage, Items.Any() ? 1 : 0);
```

Comment:
Is there any need for `Items.Any() ? 1 : 0`?
If there aren't any items, you should still be just on page 1, but with nothing displayed.
Also, you can simplify this and the next line using `Math.Clamp`.

Reply:
If no element in collection, I think you should display Page 0/0 instead of Page 1/1

Reply:
Had a crash with Math.Clamp when removing an element while being on last page

Reply:
```cs
public int CurrentPage { get; set; } = 1;
```
```cs
public int LastPage => Math.Max(1, (int)Math.Ceiling(Items.Count / (decimal)Math.Max(1, ItemsPerPage)));
```
The first Math.Max is to avoid LastPage = 0, `Math.Max(1, ItemsPerPage)` is cause apparently even if there's a minimum in the xaml, when you press down when page size is 1, it still gets set to 0 and causes an exception (it doesn't crash though, but better avoid the exception anyway).
```cs
private void UpdateCollection()
{
	CurrentPage = Math.Clamp(CurrentPage, 1, LastPage);

	DisplayedItems = Items switch
	{
		{ Count: > 0 } => Items
			.Skip(ItemsPerPage * (CurrentPage - 1))
			.Take(ItemsPerPage)
			.ToList(),

		_ => new(),
	};

	OnPropertyChanged(nameof(LastPage));
}
```
Those changes should work, no items should display page 1/1. I think it makes more sense to display that instead of 0/0.

Reply:
Changed that part